src/libraries/System.Linq/tests/SequenceTests.cs

```diff
@@ -0,0 +1,272 @@
+                {
+                    Assert.NotNull(Enumerable.Sequence(T.CreateTruncating(123), T.CreateTruncating(122), T.CreateTruncating(-i)));
+                }
+
+                ValidateThrows(T.CreateTruncating(123), T.CreateTruncating(124), T.CreateTruncating(-2));
+            }
+
+            static void ValidateUnsigned<T>() where T : INumber<T>
+            {
+                for (int i = 0; i < 3; i++)
+                {
+                    Assert.NotNull(Enumerable.Sequence(T.CreateTruncating(123), T.CreateTruncating(123), T.CreateTruncating(i)));
+                }
+
+                for (int i = 1; i < 3; i++)
+                {
+                    Assert.NotNull(Enumerable.Sequence(T.CreateTruncating(123), T.CreateTruncating(124), T.CreateTruncating(i)));
+                }
+
+                ValidateThrows(T.CreateTruncating(123), T.CreateTruncating(122), T.CreateTruncating(2));
+            }
+
+            static void ValidateThrows<T>(T start, T endInclusive, T step) where T : INumber<T>
+            {
+                AssertExtensions.Throws<ArgumentOutOfRangeException>("endInclusive", () => Enumerable.Sequence(start, endInclusive, step));
+            }
+        }
+
+        [Fact]
+        public void MultipleGetEnumeratorCalls_ReturnsUniqueInstances()
+        {
+            var sequence = Enumerable.Sequence(0, 4, 2);
+
+            using var enumerator1 = sequence.GetEnumerator();
+            using var enumerator2 = sequence.GetEnumerator();
+            Assert.NotSame(enumerator1, enumerator2);
+        }
+
+        [Fact]
+        public void Step1_MatchesRange()
+        {
+            Validate<byte>(0, 10);
+            Validate<ushort>(0, 10);
+            Validate<char>(0, 10);
+            Validate<uint>(0, 10);
+            Validate<ulong>(0, 10);
+            Validate<nuint>(0, 10);
+            Validate<UInt128>(0, 10);
+
+            Validate<sbyte>(-10, 10);
+            Validate<short>(-10, 10);
+            Validate<int>(-10, 10);
+            Validate<long>(-10, 10);
+            Validate<nint>(-10, 10);
+            Validate<Int128>(-10, 10);
+            Validate<BigInteger>(-10, 10);
+
+            Validate<Half>(-10, 10);
+            Validate<float>(-10, 10);
+            Validate<double>(-10, 10);
+
+            void Validate<T>(int startStart, int startEnd) where T : INumber<T>
+            {
+                for (int start = startStart; start <= startEnd; start++)
+                {
+                    for (int count = 1; count <= 20; count++)
+                    {
+                        Assert.Equal(
+                            Enumerable.Range(start, count).Select(i => T.CreateTruncating(i)),
+                            Enumerable.Sequence<T>(T.CreateTruncating(start), T.CreateTruncating(start + count - 1), T.One));
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void Integers_ProducesExpectedSequence()
```

Comment:
Would be nice to see a few non-generic tests that verify outputs against fixed floating point inputs (e.g. `Enumerable.Sequence(0, 1, 0.05)`), which I expect is going to be the key use case for this method.